airbyte-webapp/src/components/DeleteBlock/useDeleteModal.ts

```diff
@@ -0,0 +1,32 @@
+import { useCallback } from "react";
+import { useNavigate } from "react-router-dom";
+
+import { useConfirmationModalService } from "hooks/services/ConfirmationModal";
+
+import { DeleteBlockProps } from "./interfaces";
+
+export const useDeleteModal = ({ type, onDelete }: Partial<DeleteBlockProps>): { onDeleteButtonClick: () => void } => {
```

Comment:
I like the introduction of this hook - its a nice way to keep this logic encapsulated while allowing it to be used in either a DeleteBlock or directly by a button